src/main/java/bgu/spl/mics/MessageBrokerImpl.java

```diff
@@ -56,19 +63,24 @@ public void subscribeBroadcast(Class<? extends Broadcast> type, Subscriber m) {
 	 * @post all subscribers subscribed to b.getClass() queue's last element will be b.
 	 */
 	public void sendBroadcast(Broadcast b) {
-		// TODO Auto-generated method stub
-
+		for(Subscriber s : _broadcastSubscribers.get(b.getClass())) {
+			_subscriberQueues.get(s).add(b);
+		}
 	}
 
-	
+
 	@Override
 	/**
 	 * @pre none
 	 * @post all subscribers subscribed to e.getClass() queue's last element will be e.
 	 */
 	public <T> Future<T> sendEvent(Event<T> e) {
-		// TODO Auto-generated method stub
-		return null;
+		Subscriber currentSub = _eventSubscribers.get(e.getClass()).poll();
+		Future<T> f = new Future<>();
+		_eventFutures.put(e, f);
```

Comment:
* I think it should go to the end of the method. 
* is this `put` thread-safe? I don't really know concurrentHashMap implementation..

Reply:
- Done
- The put is thread-safe when using Concurrenthashmap, which we are